src/shell/Shell.cpp

```diff
@@ -286,11 +289,38 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
     struct RunData {
         Module* module;
         ExternVector& importValues;
-    } data = { module.value(), importValues };
+        bool hasWasiImport;
+    } data = { module.value(), importValues, hasWasiImport };
     Walrus::Trap trap;
     return trap.run([](ExecutionState& state, void* d) {
         RunData* data = reinterpret_cast<RunData*>(d);
-        data->module->instantiate(state, data->importValues);
+        Instance* instance = data->module->instantiate(state, data->importValues);
+
+        if (data->hasWasiImport) {
+            for (auto&& exp : data->module->exports()) {
+                if (exp->exportType() == ExportType::Function) {
+                    if ("_start" != exp->name()) {
+                        continue;
+                    }
+
+                    auto fn = instance->function(exp->itemIndex());
+                    FunctionType* fnType = fn->asDefinedFunction()->moduleFunction()->functionType();
+
+                    if (!fnType->param().empty()) {
+                        printf("warning: function %s has params, but params are not supported\n", exp->name().c_str());
+                        return;
+                    }
+
+                    if (!fnType->result().empty()) {
+                        printf("warning: function %s has results, but results are not supported\n", exp->name().c_str());
+                        return;
+                    }
```

Comment:
If `emcc` correctly compiled code, `_start` function would not have any arguments nor results according to the specification.
Are these checks required?
Or is there any possible case that `_start` function is arbitrarily redefined by developers?

Reply:
Yes, developers can define or redefine `_start`, thats why its was left in there. We could remove it though since the WASI standard clearly says that `_start` should not have any params / results.

Reply:
I would keep the check for safety.